src/catalog/catalog.cpp

```diff
@@ -189,507 +184,406 @@ ResultType Catalog::CreateDatabase(const std::string &database_name,
-  if (txn == nullptr) {
-    LOG_TRACE("Do not have transaction to create database: %d",
-              (int)database_oid);
-    return ResultType::FAILURE;
-  }
+  if (txn == nullptr)
+    throw CatalogException("Do not have transaction to drop database " +
+                           std::to_string(database_oid));
   auto storage_manager = storage::StorageManager::GetInstance();
   // Drop actual tables in the database
-  auto table_oids =
-      TableCatalog::GetInstance()->GetTableOids(database_oid, txn);
-  for (auto table_oid : table_oids) {
-    DropTable(database_oid, table_oid, txn);
+  auto database_object =
+      DatabaseCatalog::GetInstance()->GetDatabaseObject(database_oid, txn);
+  auto table_objects = database_object->GetTableObjects();
+  for (auto it : table_objects) {
+    DropTable(database_oid, it.second->table_oid, txn);
   }
 
   // Drop database record in catalog
-  LOG_TRACE("Deleting tuple from pg_db");
-  if (!DatabaseCatalog::GetInstance()->DeleteDatabase(database_oid, txn)) {
-    LOG_TRACE("Database tuple is not found in pg_db!");
-    return ResultType::FAILURE;
-  }
+  if (!DatabaseCatalog::GetInstance()->DeleteDatabase(database_oid, txn))
+    throw CatalogException("Database record: " + std::to_string(database_oid) +
+                           " does not exist in pg_database");
+
+  // put database object into rw_object_set
+  storage_manager->GetDatabaseWithOid(database_oid);
+  txn->RecordDrop(database_oid, INVALID_OID, INVALID_OID);
 
-  // Drop actual database object
-  LOG_TRACE("Dropping database with oid: %d", database_oid);
-  bool found_database = false;
-  std::lock_guard<std::mutex> lock(catalog_mutex);
-  found_database =
-      storage_manager->RemoveDatabaseFromStorageManager(database_oid);
-  if (!found_database) {
-    LOG_TRACE("Database %d is not found!", database_oid);
-    return ResultType::FAILURE;
-  }
   return ResultType::SUCCESS;
+  // Instead of dropping actual database object
+  // LOG_TRACE("Dropping database with oid: %d", database_oid);
+  // std::lock_guard<std::mutex> lock(catalog_mutex);
+  // bool found_database =
+  //     storage_manager->RemoveDatabaseFromStorageManager(database_oid);
+  // if (!found_database) {
+  //   LOG_TRACE("Database %d is not found!", database_oid);
+  //   return ResultType::FAILURE;
+  // }
+  // return ResultType::SUCCESS;
 }
 
 /*@brief   Drop table
-* 1. drop all the indexes on actual table, and drop index records in pg_index
-* 2. drop all the columns records in pg_attribute
-* 3. drop table record in pg_table
-* 4. delete actual table(storage level), cleanup schema, foreign keys,
-* tile_groups
-* @param   database_name    the database which the dropped table belongs to
-* @param   table_name       the dropped table name
-* @param   txn              Transaction
-* @return  Transaction ResultType(SUCCESS or FAILURE)
-*/
+ * 1. drop all the indexes on actual table, and drop index records in pg_index
```

Comment:
I noticed this at multiple instances. The only change is the additional space. I'm guessing this is a result of the clang script. Else you might want to fix it.